src/libs/workspace-utils.test.ts

```diff
@@ -158,6 +161,102 @@ describe('hasRegionConstraint', () => {
   });
 });
 
+describe('hasGroupConstraint', () => {
+  it.each([
+    { policies: [], expectedResult: false },
+    {
+      policies: [
+        {
+          namespace: 'terra',
+          name: 'group-constraint',
+          additionalData: [{ group: 'foo' }],
+        },
+      ],
+      expectedResult: true,
+    },
```

Comment:
suggested test case: the workspace has a protected-data policy but no group-constraint policies; it should still return false:
```suggestion
    },
    {
      policies: [
        {
          namespace: 'terra',
          name: 'protected-data',
          additionalData: [],
        },
      ],
      expectedResult: false,
    },
```

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/4469/commits/ef57e8e5e21382d5b59ce6cf6d6ae189e3793139.